src/access/AccessControl.cpp

```diff
@@ -23,92 +23,144 @@ namespace {
 using chip::FabricIndex;
 using namespace chip::Access;
 
-// Avoid GetAccessControl returning nullptr before SetAccessControl is called.
-class UnimplementedDataProvider : public AccessControlDataProvider
+AccessControl defaultAccessControl;
+AccessControl * globalAccessControl = &defaultAccessControl;
+
+static_assert((int(Privilege::kAdminister) & int(Privilege::kManage)) == 0);
+static_assert((int(Privilege::kAdminister) & int(Privilege::kOperate)) == 0);
+static_assert((int(Privilege::kAdminister) & int(Privilege::kView)) == 0);
+static_assert((int(Privilege::kAdminister) & int(Privilege::kProxyView)) == 0);
+static_assert((int(Privilege::kManage) & int(Privilege::kOperate)) == 0);
+static_assert((int(Privilege::kManage) & int(Privilege::kView)) == 0);
+static_assert((int(Privilege::kManage) & int(Privilege::kProxyView)) == 0);
+static_assert((int(Privilege::kOperate) & int(Privilege::kView)) == 0);
+static_assert((int(Privilege::kOperate) & int(Privilege::kProxyView)) == 0);
+static_assert((int(Privilege::kView) & int(Privilege::kProxyView)) == 0);
+
+int GetGrantedPrivileges(Privilege privilege)
```

Comment:
Suggest naming the input `aclEntryPrivilege`

Reply:
I refactored it into a function taking requestPrivilege and entryPrivilege and doing the check. That should be clear enough.